syntax = "proto3";


package noteworthy_reservation;


message ReservationRequest {
  // mydomain.noteworthy.im
  string domain = 1;
  // auth_key / invitation code
  string auth_code = 2;
}

message ReservationResponse {
  string success = 1;
  string error = 2;
}

message LinkRequest {
    // mydomain.noteworthy.im
    string domain = 1;
    // wireguard pub key
    string pub_key = 2;
    // auth_key / invitation code
    string auth_code = 3;
}

message LinkResponse {
    string link_wg_endpoint = 1;
    string link_wg_pubkey = 2;
    string link_udp_proxy_endpoint = 3;
    string link_udp_proxy_endpoint_2 = 4;
    string error = 5;
}

message RegistrationRequest {
    string somethingelse = 1; 
}

message RegistrationResponse {
    string auth_code = 1;
    string error = 2;
}